test/test.js

```diff
@@ -458,6 +462,34 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports' ||
+      pathname === '/browserWorkerTestReports') {
+
+    var map = libCoverage.createCoverageMap('window.__coverage__');
+    var summary = libCoverage.createCoverageSummary();
+
+    map.merge('self.__coverage__');
+
+    map.files().forEach(function (f) {
+      var fc = map.fileCoverageFor(f),
+        s = fc.toSummary();
+      summary.merge(s);
+    });
+
+    var writableStream = fs.createWriteStream(
+                           '../coverage/coverageinfo.json');
+
+    req.pipe(writableStream);
```

Comment:
You aren't using the `summary` variable anywhere. To use the JSON, you have to collect the full request body (using `data` and `end` events as we used before switching to `pipe`), and upon the `end` event use `JSON.parse` on the result to get back the object representation of the coverage data. This data can then be passed to `istanbul-lib-coverage`, and finally the result (`JSON.stringify(summary)` ) can be written to a file.